### PR TITLE
[568312] Project preferences and export no longer works

### PR DESCRIPTION
**Pull request by pdulth**

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/AbstractDefaultPreferencePage.java [hunk 1]
```diff
@@ -67,20 +61,7 @@ public AbstractDefaultPreferencePage(String pageId) {
...
     setDescription(getPageDescription());
   }
-
-  
-  
   
-  /**
-   * @see org.eclipse.ui.IWorkbenchPreferencePage#init(org.eclipse.ui.IWorkbench)
-   */
-  protected void setEnable(Composite parent, UserProfileModeEnum userMode) {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/AbstractDefaultPreferencePage.java:77]: useless

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/AbstractDefaultPreferencePage.java [hunk 2]
```diff
@@ -92,42 +73,16 @@ protected void setEnable(Composite parent, UserProfileModeEnum userMode) {
...
    */
   protected abstract String getPageDescription();
-
-  /**
-   * @see org.eclipse.jface.preference.FieldEditorPreferencePage#performOk()
-   */
-  @Override
-  public boolean performOk() {
-    boolean ok = super.performOk();
-    IPreferenceStore store = getPreferenceStore();
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/AbstractDefaultPreferencePage.java:102]: already done in jface

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/AbstractDefaultPreferencePage.java [hunk 3]
```diff
@@ -92,42 +73,16 @@ protected void setEnable(Composite parent, UserProfileModeEnum userMode) {
...
-
-    return ok;
-  }
   
   public boolean performCancel() {
-      
 	  return true;
   }
 
-  public IPreferenceStore getProjectPreferenceStore() {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/AbstractDefaultPreferencePage.java:123]: useless

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java [hunk 4]
```diff
@@ -19,82 +19,53 @@
...
-public abstract class ConfigurableFieldEditorPreferencePage extends FieldEditorPropertyPreferencePage implements IConfigurableFieldEditorPreferencePage,
-    IWorkbenchPropertyPage, IPageChangeProvider, IPropertyChangeListener {
+public abstract class ConfigurableFieldEditorPreferencePage extends FieldEditorPropertyPreferencePage
+    implements IConfigurableFieldEditorPreferencePage, IPageChangeProvider {
 
   private String pageId;
 
-  private IAdaptable element;
-
-  @Override
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java:57]: already done in FieldEditorPropertyPreferencePage

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java [hunk 5]
```diff
@@ -19,82 +19,53 @@
...
   }
 
   /**
-   * Creates a new field editor preference page with the given style, an empty title, and no image.
+   * Creates a new field editor preference page with the style FLAT, an empty title, and no image.
    */
   protected ConfigurableFieldEditorPreferencePage(String id) {
     super(SWT.FLAT);
     this.pageId = id;
-    getPropertyPagesIdentifients().add(pageId);
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java:96]: useless

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java [hunk 6]
```diff
@@ -105,35 +76,25 @@ public void init(IWorkbench workbench) {
     this.addPageChangedListener(new EclipseNodePreferencesChangeListener());
   }
 
-  public void propertyChange(Preferences.PropertyChangeEvent event) {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java:108]: unused

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java [hunk 7]
```diff
@@ -105,35 +76,25 @@ public void init(IWorkbench workbench) {
...
-  @Override
   protected String getPageId() {
-    return this.pageId;
-  }
-
-  /**
-   * @return
-   */
-  @Override
-  public boolean isPropertyPage() {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java:135]: already in FieldEditorPropertyPreferencePage

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java [hunk 8]
```diff
@@ -143,21 +104,21 @@ public boolean isPropertyPage() {
...
+      editor.setEnabled(
+          ConfigurabilityPreferences.isInstanceScopePreferenceItemEnabled(XmlPreferencesConfig.USER_PROFILE_MODE_ID),
+          parent);
+      parent.setEnabled(
+          ConfigurabilityPreferences.isInstanceScopePreferenceItemEnabled(XmlPreferencesConfig.USER_PROFILE_MODE_ID));
+    } else if (parent != null) {
       editor.setEnabled(true, parent);
       parent.setEnabled(true);
     }
-    ScopedCapellaPreferencesStore.fields.put(editor, null);
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java:153]: useless

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java [hunk 9]
```diff
@@ -143,21 +104,21 @@ public boolean isPropertyPage() {
...
-    editor.setPreferenceStore(Activator.getDefault().getPreferenceStore());
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java:154]: now done in setPreferenceStore

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java [hunk 10]
```diff
@@ -197,69 +157,39 @@ public void addPageChangedListener(IPageChangedListener listener) {
...
-  @Override
-  public Control getControl() {
-    return super.getControl();
-  }
+  public void enablePreferencesPage(boolean isEnable) {
 
-  @Override
-  protected IPreferenceStore doGetPreferenceStore() {
-    if (isPropertyPage()) {
-      return super.getPreferenceStore();
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java:211]: useless and risky super.getPreferenceStore call doGetPreferenceStore

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java [hunk 11]
```diff
@@ -197,69 +157,39 @@ public void addPageChangedListener(IPageChangedListener listener) {
...
+        COMPOSITE_FIELD_EDITORS.remove(i);
+      } else if (composite != null) {
+        composite.setEnabled(isEnable);
+        composite.pack();
+      }
 
-  @Override
-  public boolean performOk() {
-    super.performOk();
-    try {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/ConfigurableFieldEditorPreferencePage.java:220]: useless, we already save preference store

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/FieldEditorPropertyPreferencePage.java [hunk 12]
```diff
@@ -213,109 +177,17 @@ protected void updateFieldEditors(boolean enabled) {
...
-   */
-  @Override
-  public boolean performOk() {
-    boolean result = super.performOk();
-    if (isPropertyPage()) {
-
-      // Save state of radiobuttons in project properties
-      IResource resource = (IResource) getElement();
-      try {
-        resource.setPersistentProperty(new QualifiedName(pageId, USEPROJECTSETTINGS), TRUE);
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/FieldEditorPropertyPreferencePage.java:229]: moved to PropertyStore

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/FieldEditorPropertyPreferencePage.java [hunk 13]
```diff
@@ -213,109 +177,17 @@ protected void updateFieldEditors(boolean enabled) {
...
-
-        resource.getPersistentProperty(new QualifiedName(pageId, USEPROJECTSETTINGS));
-        if ((this.propertiesStore != null) && (this.propertiesStore instanceof IPropertyPersistentPreferenceStore)) {
-          try {
-            ((IPropertyPersistentPreferenceStore) this.propertiesStore).save();
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/FieldEditorPropertyPreferencePage.java:234]: already done in jface

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/FieldEditorPropertyPreferencePage.java [hunk 14]
```diff
@@ -213,109 +177,17 @@ protected void updateFieldEditors(boolean enabled) {
...
+      ((PropertyStore) store).setCanceled(true);
     }
-    return false;
-  }
-
-  /**
-   * Creates a new preferences page and opens it
-   * @see com.bdaum.SpellChecker.preferences.SpellCheckerPreferencePage#configureWorkspaceSettings()
-   */
-  protected void configureWorkspaceSettings() {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/FieldEditorPropertyPreferencePage.java:273]: useless

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/FieldEditorPropertyPreferencePage.java [hunk 15]
```diff
@@ -213,109 +177,17 @@ protected void updateFieldEditors(boolean enabled) {
...
-      e.printStackTrace();
-    }
-  }
-
-  /**
-   * Show a single preference pages
-   * @param id - the preference page identification
-   * @param page - the preference page
-   */
-  protected void showPreferencePage(String id, IPreferencePage page) {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/FieldEditorPropertyPreferencePage.java:293]: useless

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/FieldEditorPropertyPreferencePage.java [hunk 16]
```diff
@@ -213,109 +177,17 @@ protected void updateFieldEditors(boolean enabled) {
...
-        dialog.open();
-      }
-    });
+    return true;
   }
 
-  /**
-   * @return
-   */
-  public static Set<String> getPropertyPagesIdentifients() {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/FieldEditorPropertyPreferencePage.java:310]: useless

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/ui/AbstractCapellaCommandsPreferencePage.java [hunk 17]
```diff
@@ -12,56 +12,22 @@
...
-  @Override
-  public void init(IWorkbench workbench_p) {
-    // Nothing to do.
-  }
-
-  /**
-   * @see org.eclipse.jface.preference.FieldEditorPreferencePage#performOk()
-   */
-  @Override
-  public boolean performOk() {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/ui/AbstractCapellaCommandsPreferencePage.java:51]: already done in super

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/preferences/Activator.java [hunk 18]
```diff
@@ -384,32 +378,7 @@ public void setPropertyStore(IResource iResource, IPreferenceStore propertiesSto
    * @param capellaCommonNavigator
    */
   public void addProjectsPropertyChangeListener(IPropertyChangeListener capellaCommonNavigator) {
-
-    IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/preferences/Activator.java:388]: never occurs. 
this method is called at startup.
even if there is projects in workspace, there is no "getPropertyPreferenceStore" as it is only created when user click on Project/Properties

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/preferences/Activator.java [hunk 19]
```diff
@@ -384,32 +378,7 @@ public void setPropertyStore(IResource iResource, IPreferenceStore propertiesSto
...
-
-    }
-  }
-
-  /**
-   * @param property
-   * @return
-   */
-  @Deprecated
-  public boolean existPropertyStore(PropertyChangeEvent event) {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/preferences/Activator.java:405]: unused

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/PropertyStore.java [hunk 20]
```diff
@@ -172,18 +149,22 @@ private void writeProperties(ProjectScope project) throws IOException {
...
-   * @param value - the property value or null to delete the property
+   * 
+   * @param name
+   *          - the preference name
+   * @param value
+   *          - the property value or null to delete the property
    * @throws CoreException
    */
   private void setProperty(String name, String value) throws CoreException {
-    resource.setPersistentProperty(new QualifiedName(pageId, name), value);
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/PropertyStore.java:180]: pageId is useless and make code unclear, we don't use it afterwards in ScopedCapellaProjectScope.getProjectValue

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/preferences/CapellaPropertiesPage.java [hunk 21]
```diff
@@ -1,64 +0,0 @@
...
-
-  private IProject project;
-
-  /**
-   * @see org.eclipse.jface.preference.PreferencePage#doGetPreferenceStore()
-   */
-  @Override
-  protected IPreferenceStore doGetPreferenceStore() {
-    if (preferenceStore == null) {
-      preferenceStore = new ScopedPreferenceStore(new ProjectScope(project), CapellaActionsActivator.getDefault().getBundle().getSymbolicName());
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/preferences/CapellaPropertiesPage.java:42]: Useless, the properties page use a PropertyStore, not this one

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/preferences/DeletePropertiesPage.java [hunk 22]
```diff
@@ -1,64 +0,0 @@
...
-  private ScopedPreferenceStore preferenceStore;
-
-  private IProject project;
-
-  /**
-   * @see org.eclipse.jface.preference.PreferencePage#doGetPreferenceStore()
-   */
-  @Override
-  protected IPreferenceStore doGetPreferenceStore() {
-    if (preferenceStore == null) {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/preferences/DeletePropertiesPage.java:41]: Useless, the properties page use a PropertyStore, not this one

File: core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/PropertyStore.java [hunk 23]
```diff
@@ -26,52 +26,36 @@
...
 import org.polarsys.capella.common.tools.report.util.IReportManagerDefaultComponents;
 import org.polarsys.capella.core.preferences.Activator;
 
 /**
+ * Its overall complicated but mainly:
+ * 
+ * Unlike the default PreferenceStore, it stores all values even default ones. Indeed, Configuration projects and
+ * Capella projects can be shared between users and all preferences shall be shared between users even default ones.
+ * 
+ * It stores each preferences in the ProjectScope and as Properties using IProject.getPersistentProperty, which is idk
```
Comment (minhtutonthat) [inline core/plugins/org.polarsys.capella.core.preferences/src/org/polarsys/capella/core/commands/preferences/service/PropertyStore.java:39]: Yes, I agree that it's strange to save project preferences using both ProjectScope (saved in project .settings) and IProject.getPersistentProperty (saved in the workspace with the path to the project). But it was like this before, so I think we could keep the same code.

**Review by minhtutonthat — APPROVED** [review]

**Pull request merged**